engine/src/main/scala/cromwell/webservice/routes/wes/WesRunRoutes.scala

```diff
@@ -1,46 +1,59 @@
 package cromwell.webservice.routes.wes
 
 import akka.actor.ActorRef
-import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
-import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.model.{StatusCode, StatusCodes}
 import akka.http.scaladsl.server.Directives._
-import akka.http.scaladsl.server.Route
+import akka.http.scaladsl.server.{Directive1, Route}
 import akka.http.scaladsl.server.directives.RouteDirectives.complete
+import akka.pattern.AskTimeoutException
 import akka.util.Timeout
 import com.typesafe.config.ConfigFactory
 import cromwell.core.WorkflowId
+import cromwell.engine.workflow.WorkflowManagerActor.WorkflowNotFoundException
+import cromwell.engine.workflow.workflowstore.WorkflowStoreSubmitActor
+import cromwell.engine.workflow.workflowstore.WorkflowStoreSubmitActor.WorkflowStoreSubmitActorResponse
+import cromwell.server.CromwellShutdown
 import cromwell.services.metadata.MetadataService.{BuildMetadataJsonAction, GetSingleWorkflowMetadataAction}
 import cromwell.services.{FailedMetadataJsonResponse, SuccessfulMetadataJsonResponse}
+import cromwell.webservice.routes.CromwellApiService
 import cromwell.webservice.routes.MetadataRouteSupport.{metadataBuilderActorRequest, metadataQueryRequest}
-import cromwell.webservice.routes.wes.WesResponseJsonSupport.{WesResponseErrorFormat, WesResponseFormat}
-import cromwell.webservice.routes.wes.WesRunRoutes.{completeCromwellResponse, runLog}
+import cromwell.webservice.routes.wes.WesRunRoutes.{WesErrorHandler, WesSuccessHandler, extractSubmission, completeCromwellResponse, runLog}
 import net.ceedubs.ficus.Ficus._
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 import scala.concurrent.duration.FiniteDuration
 import scala.util.{Failure, Success}
 
-trait WesRunRoutes {
+trait WesRunRoutes extends CromwellApiService {
```

Comment:
At a high level, this does not seem right to me. I think `CromwellApiService` should be a sibling to `WesRunRoutes` and code that needs to be shared should be extracted to a third location such as a utility trait.

As an example, observe how `WebServiceUtils` is mixed in to both `CromwellApiService` and `WomtoolRouteSupport`. As a matter of fact, `WebServiceUtils` may be the right place for said shared code.

Reply:
Just checking my understanding, so we want to keep from using `extend CromwellApiService` and instead we take the shared code which is the `submitRequest` function, and move that function into `WebServiceUtils` since it is now being shared among several files. Is that the right thinking?